### ISSUE TITLE
IDE does not start: "EADDRNOTAVAIL: address not available"

### ISSUE THREAD

**Issue by sclsj** [arduino/arduino-ide#2536]
### Describe the problem

IDE starts, but does not present the main window

```
jin@Joy-MBP ~> "/Applications/Arduino IDE.app/Contents/MacOS/Arduino IDE"
Arduino IDE 2.3.3
Checking for frontend application configuration customizations. Module path: /Applications/Arduino IDE.app/Contents/Resources/app/lib/backend/electron-main.js, destination 'package.json': /Applications/Arduino IDE.app/Contents/Resources/app/package.json
Setting 'theia.frontend.config.appVersion' application configuration value to: "2.3.3" (type of string)
Setting 'theia.frontend.config.cliVersion' application configuration value to: "1.0.4" (type of string)
Setting 'theia.frontend.config.buildDate' application configuration value to: "2024-09-25T09:35:36.798Z" (type of string)
Frontend application configuration after modifications: {"applicationName":"Arduino IDE","defaultTheme":{"light":"arduino-theme","dark":"arduino-theme-dark"},"defaultIconTheme":"none","electron":{"windowOptions":{}},"defaultLocale":"","validatePreferencesSchema":false,"preferences":{"window.title":"${rootName}${activeEditorShort}${appName}","files.autoSave":"afterDelay","editor.minimap.enabled":false,"editor.tabSize":2,"editor.scrollBeyondLastLine":false,"editor.quickSuggestions":{"other":false,"comments":false,"strings":false},"editor.maxTokenizationLineLength":500,"editor.bracketPairColorization.enabled":false,"breadcrumbs.enabled":false,"workbench.tree.renderIndentGuides":"none","explorer.compactFolders":false},"appVersion":"2.3.3","cliVersion":"1.0.4","buildDate":"2024-09-25T09:35:36.798Z"}
Starting backend process. PID: 5292
Using browser-only version of superagent in non-browser environment
Configuration directory URI: 'file:///Users/jin/.arduinoIDE'
Configuring to accept webviews on '^.+\.webview\..+$' hostname.
2024-10-22T01:52:38.564Z root INFO Backend Object.initialize: 1.9 ms [Finished 0.273 s after backend start]
2024-10-22T01:52:38.564Z root INFO Backend Object.configure: 1.4 ms [Finished 0.273 s after backend start]
2024-10-22T01:52:38.564Z root INFO Backend a.onStart: 0.7 ms [Finished 0.273 s after backend start]
2024-10-22T01:52:38.565Z root INFO Backend a.configure: 0.1 ms [Finished 0.286 s after backend start]
2024-10-22T01:52:38.565Z root INFO Backend d.onStart: 0.1 ms [Finished 0.286 s after backend start]
2024-10-22T01:52:38.566Z root INFO Backend d.configure: 0.1 ms [Finished 0.286 s after backend start]
2024-10-22T01:52:38.566Z root INFO Backend d.initialize: 13.0 ms [Finished 0.286 s after backend start]
2024-10-22T01:52:38.566Z root INFO Backend c.onStart: 0.1 ms [Finished 0.286 s after backend start]
2024-10-22T01:52:38.566Z root INFO Backend s.configure: 0.5 ms [Finished 0.287 s after backend start]
2024-10-22T01:52:38.567Z config INFO >>> Initializing CLI configuration...
2024-10-22T01:52:38.567Z root INFO Backend c.initialize: 0.2 ms [Finished 0.287 s after backend start]
2024-10-22T01:52:38.567Z root INFO Backend x.onStart: 0.4 ms [Finished 0.287 s after backend start]
2024-10-22T01:52:38.567Z config INFO Loading CLI configuration from /Users/jin/.arduinoIDE/arduino-cli.yaml...
2024-10-22T01:52:38.567Z daemon INFO Starting daemon from /Applications/Arduino IDE.app/Contents/Resources/app/lib/backend/resources/arduino-cli...
2024-10-22T01:52:38.567Z root INFO Backend s.initialize: 0.2 ms [Finished 0.288 s after backend start]
2024-10-22T01:52:38.567Z root INFO Backend w.onStart: 0.5 ms [Finished 0.288 s after backend start]
2024-10-22T01:52:38.567Z root INFO Backend r.configure: 0.9 ms [Finished 0.288 s after backend start]
2024-10-22T01:52:38.567Z discovery-log INFO start
2024-10-22T01:52:38.567Z discovery-log INFO start new deferred
2024-10-22T01:52:38.567Z root INFO Backend d.initialize: 0.2 ms [Finished 0.288 s after backend start]
2024-10-22T01:52:38.567Z root INFO Backend y.onStart: 0.8 ms [Finished 0.289 s after backend start]
2024-10-22T01:52:38.567Z root INFO Backend r.configure: 0.7 ms [Finished 0.289 s after backend start]
2024-10-22T01:52:38.567Z root INFO Backend s.onStart: 0.2 ms [Finished 0.289 s after backend start]
2024-10-22T01:52:38.567Z root INFO Backend h.configure: 0.2 ms [Finished 0.289 s after backend start]
2024-10-22T01:52:38.567Z root INFO Backend a.configure: 0.0 ms [Finished 0.289 s after backend start]
2024-10-22T01:52:38.567Z root INFO Backend l.configure: 0.0 ms [Finished 0.290 s after backend start]
2024-10-22T01:52:38.569Z root ERROR Failed to start the backend application:
2024-10-22T01:52:38.569Z root ERROR Error: listen EADDRNOTAVAIL: address not available 10.53.146.123
    at Server.setupListenHandle [as _listen2] (node:net:1734:21)
    at listenInCluster (node:net:1799:12)
    at GetAddrInfoReqWrap.doListen [as callback] (node:net:1948:7)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:110:8)
2024-10-22T01:52:38.569Z root ERROR Uncaught Exception:  Error: listen EADDRNOTAVAIL: address not available 10.53.146.123
2024-10-22T01:52:38.569Z root ERROR Error: listen EADDRNOTAVAIL: address not available 10.53.146.123
    at Server.setupListenHandle [as _listen2] (node:net:1734:21)
    at listenInCluster (node:net:1799:12)
    at GetAddrInfoReqWrap.doListen [as callback] (node:net:1948:7)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:110:8)
2024-10-22T01:52:38.571Z root WARN The local plugin referenced by local-dir:/Users/jin/.arduinoIDE/plugins does not exist.
2024-10-22T01:52:38.571Z root WARN The local plugin referenced by local-dir:/Users/jin/.arduinoIDE/extensions does not exist.

^CStored workspaces roots: 
No sketches were scheduled for deletion.
Could not terminate the backend process. It was not running.
```

### To reproduce

Open the application

### Expected behavior

Main window opens

### Arduino IDE version

Lastest nightly build and lastest stable, also tested local compile (git head)

### Operating system

macOS

### Operating system version

12.6

### Additional context

Tried removing `~/.arduinoIDE`

```
ifconfig
lo0: flags=8049<UP,LOOPBACK,RUNNING,MULTICAST> mtu 16384
	options=1203<RXCSUM,TXCSUM,TXSTATUS,SW_TIMESTAMP>
	inet6 ::1 prefixlen 128 
	inet6 fe80::1%lo0 prefixlen 64 scopeid 0x1 
	nd6 options=201<PERFORMNUD,DAD>
gif0: flags=8010<POINTOPOINT,MULTICAST> mtu 1280
stf0: flags=0<> mtu 1280
anpi0: flags=8822<BROADCAST,SMART,SIMPLEX,MULTICAST> mtu 1500
	options=400<CHANNEL_IO>
	ether 1a:8b:30:23:34:7c 
	inet6 fe80::188b:30ff:fe23:347c%anpi0 prefixlen 64 scopeid 0x4 
	nd6 options=201<PERFORMNUD,DAD>
	media: none
	status: inactive
anpi2: flags=8822<BROADCAST,SMART,SIMPLEX,MULTICAST> mtu 1500
	options=400<CHANNEL_IO>
	ether 1a:8b:30:23:34:7e 
	inet6 fe80::188b:30ff:fe23:347e%anpi2 prefixlen 64 scopeid 0x5 
	nd6 options=201<PERFORMNUD,DAD>
	media: none
	status: inactive
anpi1: flags=8822<BROADCAST,SMART,SIMPLEX,MULTICAST> mtu 1500
	options=400<CHANNEL_IO>
	ether 1a:8b:30:23:34:7d 
	inet6 fe80::188b:30ff:fe23:347d%anpi1 prefixlen 64 scopeid 0x6 
	nd6 options=201<PERFORMNUD,DAD>
	media: none
	status: inactive
en4: flags=8822<BROADCAST,SMART,SIMPLEX,MULTICAST> mtu 1500
	options=400<CHANNEL_IO>
	ether 1a:8b:30:23:34:5c 
	nd6 options=201<PERFORMNUD,DAD>
	media: none
	status: inactive
en5: flags=8822<BROADCAST,SMART,SIMPLEX,MULTICAST> mtu 1500
	options=400<CHANNEL_IO>
	ether 1a:8b:30:23:34:5d 
	nd6 options=201<PERFORMNUD,DAD>
	media: none
	status: inactive
en6: flags=8822<BROADCAST,SMART,SIMPLEX,MULTICAST> mtu 1500
	options=400<CHANNEL_IO>
	ether 1a:8b:30:23:34:5e 
	nd6 options=201<PERFORMNUD,DAD>
	media: none
	status: inactive
en1: flags=8963<UP,BROADCAST,SMART,RUNNING,PROMISC,SIMPLEX,MULTICAST> mtu 1500
	options=460<TSO4,TSO6,CHANNEL_IO>
	ether 36:42:ab:3f:93:80 
	media: autoselect <full-duplex>
	status: inactive
en2: flags=8963<UP,BROADCAST,SMART,RUNNING,PROMISC,SIMPLEX,MULTICAST> mtu 1500
	options=460<TSO4,TSO6,CHANNEL_IO>
	ether 36:42:ab:3f:93:84 
	media: autoselect <full-duplex>
	status: inactive
en3: flags=8963<UP,BROADCAST,SMART,RUNNING,PROMISC,SIMPLEX,MULTICAST> mtu 1500
	options=460<TSO4,TSO6,CHANNEL_IO>
	ether 36:42:ab:3f:93:88 
	media: autoselect <full-duplex>
	status: inactive
ap1: flags=8802<BROADCAST,SIMPLEX,MULTICAST> mtu 1500
	options=400<CHANNEL_IO>
	ether f2:2f:4b:13:be:76 
	media: autoselect
	status: inactive
en0: flags=8863<UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 1500
	options=6463<RXCSUM,TXCSUM,TSO4,TSO6,CHANNEL_IO,PARTIAL_CSUM,ZEROINVERT_CSUM>
	ether f0:2f:4b:13:be:76 
	inet6 fe80::1c9a:b0e9:e2da:85dd%en0 prefixlen 64 secured scopeid 0xf 
	inet 10.53.20.82 netmask 0xffffe000 broadcast 10.53.31.255
	nd6 options=201<PERFORMNUD,DAD>
	media: autoselect
	status: active
bridge0: flags=8822<BROADCAST,SMART,SIMPLEX,MULTICAST> mtu 1500
	options=63<RXCSUM,TXCSUM,TSO4,TSO6>
	ether 36:42:ab:3f:93:80 
	Configuration:
		id 0:0:0:0:0:0 priority 0 hellotime 0 fwddelay 0
		maxage 0 holdcnt 0 proto stp maxaddr 100 timeout 1200
		root id 0:0:0:0:0:0 priority 0 ifcost 0 port 0
		ipfilter disabled flags 0x0
	member: en1 flags=3<LEARNING,DISCOVER>
	        ifmaxaddr 0 port 11 priority 0 path cost 0
	member: en2 flags=3<LEARNING,DISCOVER>
	        ifmaxaddr 0 port 12 priority 0 path cost 0
	member: en3 flags=3<LEARNING,DISCOVER>
	        ifmaxaddr 0 port 13 priority 0 path cost 0
	media: <unknown type>
	status: inactive
awdl0: flags=8943<UP,BROADCAST,RUNNING,PROMISC,SIMPLEX,MULTICAST> mtu 1500
	options=400<CHANNEL_IO>
	ether de:69:b1:03:e3:bf 
	inet6 fe80::dc69:b1ff:fe03:e3bf%awdl0 prefixlen 64 scopeid 0x11 
	nd6 options=201<PERFORMNUD,DAD>
	media: autoselect
	status: active
llw0: flags=8863<UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 1500
	options=400<CHANNEL_IO>
	ether de:69:b1:03:e3:bf 
	inet6 fe80::dc69:b1ff:fe03:e3bf%llw0 prefixlen 64 scopeid 0x12 
	nd6 options=201<PERFORMNUD,DAD>
	media: autoselect
	status: active
utun0: flags=8051<UP,POINTOPOINT,RUNNING,MULTICAST> mtu 1380
	inet6 fe80::ad1b:2a7a:6faf:95dc%utun0 prefixlen 64 scopeid 0x13 
	nd6 options=201<PERFORMNUD,DAD>
utun1: flags=8051<UP,POINTOPOINT,RUNNING,MULTICAST> mtu 2000
	inet6 fe80::e0e8:68ab:1dc9:ce2f%utun1 prefixlen 64 scopeid 0x14 
	nd6 options=201<PERFORMNUD,DAD>
utun2: flags=8051<UP,POINTOPOINT,RUNNING,MULTICAST> mtu 1000
	inet6 fe80::ce81:b1c:bd2c:69e%utun2 prefixlen 64 scopeid 0x15 
	nd6 options=201<PERFORMNUD,DAD>
utun3: flags=8051<UP,POINTOPOINT,RUNNING,MULTICAST> mtu 1380
	inet6 fe80::3365:a83d:1120:8c06%utun3 prefixlen 64 scopeid 0x18 
	nd6 options=201<PERFORMNUD,DAD>
utun4: flags=8051<UP,POINTOPOINT,RUNNING,MULTICAST> mtu 1380
	inet6 fe80::37e4:792f:210b:b2a0%utun4 prefixlen 64 scopeid 0x19 
	nd6 options=201<PERFORMNUD,DAD>
vmenet0: flags=8963<UP,BROADCAST,SMART,RUNNING,PROMISC,SIMPLEX,MULTICAST> mtu 1500
	ether 8e:f4:2e:cd:93:f9 
	media: autoselect
	status: active
bridge100: flags=8822<BROADCAST,SMART,SIMPLEX,MULTICAST> mtu 1500
	options=3<RXCSUM,TXCSUM>
	ether f2:2f:4b:31:9d:64 
	inet 10.211.55.2 netmask 0xffffff00 broadcast 10.211.55.255
	inet6 fe80::f02f:4bff:fe31:9d64%bridge100 prefixlen 64 scopeid 0x1b 
	inet6 fdb2:2c26:f4e4::1 prefixlen 64 
	Configuration:
		id 0:0:0:0:0:0 priority 0 hellotime 0 fwddelay 0
		maxage 0 holdcnt 0 proto stp maxaddr 100 timeout 1200
		root id 0:0:0:0:0:0 priority 0 ifcost 0 port 0
		ipfilter disabled flags 0x0
	member: vmenet0 flags=3<LEARNING,DISCOVER>
	        ifmaxaddr 0 port 26 priority 0 path cost 0
	nd6 options=201<PERFORMNUD,DAD>
	media: autoselect
	status: active
vmenet1: flags=8963<UP,BROADCAST,SMART,RUNNING,PROMISC,SIMPLEX,MULTICAST> mtu 1500
	ether f2:df:68:29:46:08 
	media: autoselect
	status: active
bridge101: flags=8822<BROADCAST,SMART,SIMPLEX,MULTICAST> mtu 1500
	options=3<RXCSUM,TXCSUM>
	ether f2:2f:4b:31:9d:65 
	inet 10.37.129.2 netmask 0xffffff00 broadcast 10.37.129.255
	inet6 fe80::f02f:4bff:fe31:9d65%bridge101 prefixlen 64 scopeid 0x1d 
	inet6 fdb2:2c26:f4e4:1::1 prefixlen 64 
	Configuration:
		id 0:0:0:0:0:0 priority 0 hellotime 0 fwddelay 0
		maxage 0 holdcnt 0 proto stp maxaddr 100 timeout 1200
		root id 0:0:0:0:0:0 priority 0 ifcost 0 port 0
		ipfilter disabled flags 0x0
	member: vmenet1 flags=3<LEARNING,DISCOVER>
	        ifmaxaddr 0 port 28 priority 0 path cost 0
	nd6 options=201<PERFORMNUD,DAD>
	media: autoselect
	status: active
en10: flags=8863<UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 1500
	options=400<CHANNEL_IO>
	ether d2:d1:71:23:09:95 
	inet6 fe80::d0d1:71ff:fe23:995%en10 prefixlen 64 scopeid 0x7 
	nd6 options=201<PERFORMNUD,DAD>
	media: autoselect
	status: active
```

### Issue checklist

- [x] I searched for previous reports in [the issue tracker](https://github.com/arduino/arduino-ide/issues?q=)
- [x] I verified the problem still occurs when using the latest [nightly build](https://www.arduino.cc/en/software#nightly-builds)
- [x] My report contains all necessary details

**Comment by per1234** [comment]
Hi @sclsj. This is caused by a problem with your system, rather than Arduino IDE. I think it will be more appropriate for us to assist you in solving the problem over on Arduino Forum:

https://forum.arduino.cc/c/software/ide-2-x/93

Make a forum topic and we'll help you out over there.

**Comment by sclsj** [comment]
I'm having some trouble signing up for the Arduino Forum. Can you elaborate why you determined that this is caused by a problem with my system?

**Comment by sclsj** [comment]
Just let it sit there for 10 minutes and it suddenly went through. 

https://forum.arduino.cc/t/ide-does-not-start-eaddrnotavail-address-not-available/1313854
